packages/preferences/src/browser/ui/settings-editor.tsx

```diff
@@ -0,0 +1,542 @@
+/********************************************************************************
+ * Copyright (C) 2020 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { postConstruct, injectable, inject } from 'inversify';
+import {
+    ReactWidget,
+    PreferenceService,
+    PreferenceSchemaProvider,
+    PreferenceDataProperty,
+    JsonType,
+    PreferenceScope,
+} from '@theia/core/lib/browser';
+import * as React from 'react';
+import { MessageService } from '@theia/core';
+import { PreferenceConfigurations } from '@theia/core/lib/browser/preferences/preference-configurations';
+import { debounce, isEmpty, pickBy } from 'lodash';
+import { SettingsFuzzySearch } from './settings-fuzzy-search';
+import { SettingsObjectInput } from './settings-object-input';
+import { SettingsArray } from './settings-array';
+import { SettingsNavbar } from './settings-navbar';
+import { SettingsPreferenceContextMenuWidget } from './settings-preference-context-menu-widget';
+
+@injectable()
+export class SettingsEditor extends ReactWidget {
+    static readonly ID = 'settings.editor';
+    static readonly LABEL = 'Settings Editor';
+
+    readonly scopes: PreferenceScope[] = PreferenceScope.getScopes().slice(1, 3);
+
+    properties: any;
```

Comment:
Should be everything is expose to clients as public and be mutable?

Reply:
I have marked everything I deemed to not need to be exposed to clients in the file with "protected."